recommend/admissionControllerPolicy.go

```diff
@@ -0,0 +1,92 @@
+package recommend
+
+import (
+	"context"
+	"errors"
+	"github.com/accuknox/auto-policy-discovery/src/libs"
+	"github.com/accuknox/auto-policy-discovery/src/protobuf/v1/worker"
+	"github.com/clarketm/json"
+	"github.com/fatih/color"
+	"github.com/kubearmor/kubearmor-client/k8s"
+	"github.com/kubearmor/kubearmor-client/utils"
+	kyvernov1 "github.com/kyverno/kyverno/api/kyverno/v1"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/exp/slices"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"os"
+	"strconv"
+	"strings"
+)
+
+func getClientConnection(c *k8s.Client) (*grpc.ClientConn, error) {
+	gRPC := ""
+	targetSvc := "discovery-engine"
+	var port int64 = 9089
+	mtchLabels := map[string]string{"app": "discovery-engine"}
+	if val, ok := os.LookupEnv("DISCOVERY_SERVICE"); ok {
+		gRPC = val
+	} else {
+		pf, err := utils.InitiatePortForward(c, port, port, mtchLabels, targetSvc)
+		if err != nil {
+			return nil, err
+		}
+		gRPC = "localhost:" + strconv.FormatInt(pf.LocalPort, 10)
+	}
+	// create a client
+	conn, err := grpc.Dial(gRPC, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		return nil, errors.New("could not connect to the server. Possible troubleshooting:\n- Check if discovery engine is running\n- Create a portforward to discovery engine service using\n\t\033[1mkubectl port-forward -n explorer service/knoxautopolicy --address 0.0.0.0 --address :: 9089:9089\033[0m\n[0m")
+	}
+	return conn, nil
+}
+
+func recommendAdmissionControllerPolicies(img ImageInfo, c *k8s.Client) error {
+	conn, err := getClientConnection(c)
+	if err != nil {
+		return err
```

Comment:
We will be creating a new client connection and port forward for each image we have. Ideally we should have persistent connection with `discovery-engine` while generate report for all images.

Reply:
Yes, that is better, changed to have a single connection.